controllers/appgroup_controller.go

```diff
@@ -153,7 +153,7 @@ func (r *ApplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	// If the (needs) Rollback phase is present in the reconciled version,
 	// we must rollback the application group to the last successful spec.
 	// This should only happen on updates and not during installs.
-	if appGroup.GetDeployCondition() == meta.RollingBackReason {
+	if appGroup.GetDeployCondition() == meta.RollingBackReason && r.lastSuccessfulApplicationGroup != nil {
```

Comment:
If the first install fails, it will continue to install the object over and over again with no backoff, maybe we should consider having some backoff logic for failure?

Reply:
Do you mean during rollback or the first time (helm install). If it errors, do we not just drop it and not reconcile after cleaning up?

Reply:
During rollback, after there is a failure on install. If this check fails, won't it continue to go down into the other conditions below where it will attempt to re-install again?

Reply:
Added an if else block